src/main/scala/org/hammerlab/guacamole/reads/MappedRead.scala

```diff
@@ -102,6 +102,63 @@ case class MappedRead(
     (pos, element) => pos + element.getLength
   })
 
+  /**
+   * Take a subsequence or slice of a read [from, until)
+   * Returns None if the provided coordinates do not overlap the read
+   *
+   * @param from reference locus to start the new read
+   * @param until reference locus to slice up to (exclusive end)
+   * @param referenceContigSequence reference sequence for the read's contig
+   * @return A read which spans [from, until) or None if the provided coordinates do not overlap the read
+   */
+  def slice(from: Long, until: Long, referenceContigSequence: ContigSequence): Option[MappedRead] = {
+    if (from >= end || until < start) {
+      None
+    } else if (from <= start && until >= end) {
+      Some(this)
+    } else {
+
+      val referenceStart = math.max(from, start)
+      val referenceEnd = math.min(until - 1, end - 1)
+
+      val el = PileupElement(this, start, referenceContigSequence)
+      val startElement = el.advanceToLocus(referenceStart)
```

Comment:
I don't love having to instantiate a PileupElement to get this functionality; the dependency graph of the project should generally not have Reads depending on Pileup things, I think. Might make sense to pull this functionality out or put it into MappedRead and let PileupElement get it from there?